example_js_standalone_smart_card_client_app/background.js

```diff
@@ -87,6 +87,49 @@ function listReaders() {
 /** @param {!Array.<string>} readers List of reader names. */
 function onReadersListed(readers) {
   console.log('List of PC/SC-Lite readers: ' + readers);
+  if (readers.length)
+    getReaderState(readers[0]);
+}
+
+/** @param {string} reader Name of the reader */
+function getReaderState(reader) {
+  console.log('Obtaining reader state');
+  api.SCardGetStatusChange(
+         sCardContext, GoogleSmartCard.PcscLiteClient.API.INFINITE, [{
+           reader_name: reader,
+           current_state: GoogleSmartCard.PcscLiteClient.API.SCARD_STATE_UNAWARE
+         }])
+      .then(function(result) {
+        result.get(onReaderStateGot, onPcscLiteError);
+      }, onRequestFailed);
+}
+
+/**
+ * @param {!Array.<!GoogleSmartCard.PcscLiteClient.API.SCARD_READERSTATE_OUT>}
+ *     readerStates
+ */
+function onReaderStateGot(readerStates) {
+  console.log(
+      'Waiting for reader state change. Current state: ' +
+      JSON.stringify(readerStates));
+  api.SCardGetStatusChange(
+         sCardContext, GoogleSmartCard.PcscLiteClient.API.INFINITE, [{
+           reader_name: readerStates[0].reader_name,
```

Comment:
Yes, exactly one, because that's how many readers we passed to `SCardGetStatusChange()` in `getReaderState()`. This contract was more obvious in the original C interface, because in C both inputs and outputs are passed via a single argument `rgReaderStates`, which is a pointer.